### PR TITLE
Indexer

### PR DESCRIPTION
**Pull request by Rohan-Sreenivasan**
It can index the balls

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -15,5 +14,6 @@
     "**/.project": true,
     "**/.settings": true,
     "**/.factorypath": true
-  }
+  },
+  "workbench.colorTheme": "Visual Studio Dark"
```
Comment (bot190) [inline .vscode/settings.json]: This file should not be included in the PR.

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -46,13 +51,14 @@
     final private NetworkTableEntry nameEntry = NetworkTableInstance.getDefault().getEntry("RobotName");
     final private String robotName = nameEntry.getString("Unknown");
 
-    final private RobotMap map = RobotUtils.getMapForName(robotName, RobotMap.class, "frc.robot.maps", new RobotMap());
+    final private RobotMap map = new TempestMap();
```
Comment (bot190) [inline src/main/java/frc/robot/Robot.java]: You'll need to pull in the changes that @msoucy and @jeffaburke  made to this mechanism and this line should be reverted.

File: src/main/java/frc/robot/maps/FrancoisMap.java [hunk 3]
```diff
@@ -57,7 +57,7 @@ public WDSolenoid deployIntake() {
     public ShooterMap getShooterMap() {
         return new ShooterMap() {
             @Override
-            public PIDSparkMax shooterWheel() {
+            public PIDSparkMax shooterWheel1() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/FrancoisMap.java]: Seems like there's an errant `1` in here.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 4]
```diff
@@ -30,21 +35,36 @@ private EncodedSpeedController getMock() {
...
+
     }
 
     public IntakeMap getIntakeMap() {
         return new IntakeMap();
     }
 
     public static class ShooterMap {
-        public SendableSpeedController shooterWheel() {
+        public SendableSpeedController shooterWheel1() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: There shouldn't be any shooter changes in this PR. Seems like there might be some weird merge issue.

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 5]
```diff
@@ -73,4 +93,40 @@ public LiftMap getLiftMap() {
...
+        public BooleanSupplier topPierreIR() {
+            return () -> false;
+        }
+
+        public BooleanSupplier backIntakeIR() {
+            return () -> false;
+        }
+
+
+        // AnalogTrigger frontIntakeIR = new AnalogTrigger(0);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/RobotMap.java]: Please remove these commented lines.

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 6]
```diff
@@ -1,32 +1,121 @@
...
+import edu.wpi.first.wpilibj.Victor;
+
 @RobotMapFor("Tempest")
 public class TempestMap extends RobotMap {
 
     @Override
     public DifferentialDriveMap getDriveMap() {
         return new DifferentialDriveMap() {
+            @Override
+            public EncodedSpeedController getLeft() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java:29]: It looks like these are mostly formatting changes, but they probably shouldn't be here either.

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 7]
```diff
@@ -1,32 +1,121 @@
...
+            public BooleanSupplier frontIntakeIR() {
+                AnalogTrigger frontIntakeIR = new AnalogTrigger(0);
+                frontIntakeIR.setLimitsVoltage(1.2,1.4);
+                return frontIntakeIR::getTriggerState;
             }
+
+            public BooleanSupplier bottomPierreIR() {
+                AnalogTrigger bottomPierreIR = new AnalogTrigger(1);
+                bottomPierreIR.setLimitsVoltage(1.2, 1.4);
+                return bottomPierreIR::getTriggerState;            }
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java]: Not sure why this file didn't autoformat, the `}` should be on a new line.

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 8]
```diff
@@ -1,32 +1,121 @@
...
+
+            public BooleanSupplier topPierreIR() {
+                AnalogTrigger topPierreIR = new AnalogTrigger(2);
+                topPierreIR.setLimitsVoltage(1.2, 1.4);
+                return topPierreIR::getTriggerState;            }
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java]: Same here.

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 9]
```diff
@@ -1,32 +1,121 @@
...
+
+            public BooleanSupplier backIntakeIR() {
+                AnalogTrigger backIntakeIR = new AnalogTrigger(3);
+                backIntakeIR.setLimitsVoltage(1.2, 1.4);
+                return backIntakeIR::getTriggerState;
+            }
+
+
+    
+            // AnalogTrigger frontIntakeIR = new AnalogTrigger(0);
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java]: Please remove these commented lines (and the blank lines around them).

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 10]
```diff
@@ -1,32 +1,121 @@
...
+            // AnalogTrigger bottomPierreIR = new AnalogTrigger(1);
+            // AnalogTrigger topPierreIR = new AnalogTrigger(2);
+            // AnalogTrigger backIntakeIR = new AnalogTrigger(3);
+
+        };
+    }
+
+    @Override
+    public ShooterMap getShooterMap() {
```
Comment (bot190) [inline src/main/java/frc/robot/maps/TempestMap.java:87]: Again, not sure why these are showing up here, but there shouldn't be shooter stuff in this PR.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+
+    public Indexer(final IndexMap map) {
+        super();
+        singulator = map.singulator();
+        frontIntakeIR = map.frontIntakeIR();
+        bottomPierreIR = map.bottomPierreIR();
+        topPierreIR = map.topPierreIR();
+        backIntakeIR = map.backIntakeIR();
+
+        // bottomPierreIR = map.bottomPierreIR();
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Please remove these commented lines.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+
+    public SequentialCommandGroup intakeToPierre() {
+        return new SequentialCommandGroup(pierrePossesion(), runToClear());
+    }
+    
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered(){
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: What is this command group doing? It seems like it would intake a ball and immediately shoot it.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        });
+    }
+
+    public CommandBase runPierre() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: This isn't used in this file, and should probably be removed. We shouldn't need a command like this.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+    public CommandBase quicklyOutput() {
+        return indexMotor(pierreIndexSpeed);
+    }
+
+    public CommandBase reverseOutput() {
+        return indexMotor(-pierreIndexSpeed);
+    }
+
+    /*
+     * Command to run the singlator motor until the IR sensor 1 (analog 0) Command
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: It looks like this first sentence ends prematurely. Use the new sensor names instead of `IR senor 1` (there's also a spelling mistake).

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+            {
+                addRequirements(Indexer.this);
+            }
+
+
+            @Override
+            public boolean isFinished() {
+                return frontIntakeIR.getAsBoolean();
+                
+                //frontIntakeIR.getVoltage() > irSensorVoltage;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: These lines can be removed.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+                // values of .8 when open, 1.7 when closed
+
+            }
+
+            @Override
+            public void execute() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: I believe you can skip defining this function if it's empty.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 17]
```diff
@@ -0,0 +1,335 @@
...
+
+            @Override
+            public void end(final boolean interrupted) {
+                singulator.set(0);
+            }
+
+        };
+
+    }
+    //This command will make sure that the singulator has possesion of the ball
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Comments should go before the code they reference or on the same line, not afterwards.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 18]
```diff
@@ -0,0 +1,335 @@
...
+
+    public CommandBase pierrePossesion() {
+        return new CommandBase() {
+
+                @Override
+                public void initialize() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Since we end up doing the same thing in `execute` we can probably get rid of this function entirely.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 19]
```diff
@@ -0,0 +1,335 @@
...
+                if (backIntakeIR.getAsBoolean()) {
+                    pierreMotor.set(pierreIndexSpeed);
+                    singulator.set(singulatorMotorSpeed);
+                }
+            }
+             
+
+            public boolean isFinished() {
+                return bottomPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: This comment is no longer relevant here. Please remove it.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 20]
```diff
@@ -0,0 +1,335 @@
...
+                pierreMotor.set(0);
+               
+            }
+
+        };
+
+    }
+    //this will bring the ball to the top
+    
+    public CommandBase unLoadBall() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java:137]: We should mention that a ball must already be at the top of Pierre for this command to work.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 21]
```diff
@@ -0,0 +1,335 @@
...
+                SmartDashboard.putNumber("Ball Count", ballCounting);
+
+            }
+
+        };
+
+    }
+    //this will bring the ball to the shooter
+
+    public CommandBase runToClear() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: This should be `runToClearBottomSensor` or something.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 22]
```diff
@@ -0,0 +1,335 @@
...
+
+            @Override
+            public boolean isFinished() {
+                return !bottomPierreIR.getAsBoolean();
+                // values of .8 when open, 2.4 when closed
+
+            }
+
+            @Override
+            public void execute() {
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: This function can be removed entirely.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 23]
```diff
@@ -0,0 +1,335 @@
...
+                // pierreMotor.set(0);
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                ballCounting++;
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java:171]: We should only do this if the command is not interrupted.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 24]
```diff
@@ -0,0 +1,335 @@
...
+
+            @Override
+            public void execute() {
+
+            }
+
+            @Override
+            public void end(final boolean interrupted) {
+                pierreMotor.set(0);
+                singulator.set(0);
```
Comment (bot190) [inline src/main/java/frc/robot/subsystems/Indexer.java]: These are unnecessary as the running command should also stop the motor in its own `end`

**Review by bot190 — CHANGES_REQUESTED** [review]
Please remove the extra newlines, there should be one between each function and class, but there's no reason for whitespace within the functions.

Instead of a `ParallelDeadlineGroup` with all of the commands, we want separate `ParallelDeadlineGroup` for each of the commands that run Pierre, We would then use those `ParallelDeadlineGroup` to ensure that we don't stuff a ball into the shooter. We can talk about this tomorrow to clarify.

These commands are all quite small, they could probably be implemented with the `FunctionalCommand` helper and take up much less space.

File: .vscode/settings.json [hunk 25]
```diff
@@ -1,6 +1,5 @@
 {
   "java.configuration.updateBuildConfiguration": "automatic",
-  "editor.formatOnSave": true,
```
Comment (msoucy) [inline .vscode/settings.json]: We want this line. Please re-add it.

File: src/main/java/frc/robot/maps/TempestMap.java [hunk 26]
```diff
@@ -1,32 +1,114 @@
...
+
+
+            }
 
             @Override
             public EncodedSpeedController getRight() {
-                final SendableSpeedController rightGroup = SendableSpeedController.group(new WPI_TalonSRX(2),
-                        new WPI_TalonSRX(3));
-                return EncodedSpeedController.join(rightGroup, new MockEncoder());
+                SendableSpeedController motors = SendableSpeedController
```
Comment (msoucy) [inline src/main/java/frc/robot/maps/TempestMap.java]: Instead of calling `SendableSpeedController.wrap(new SpeedControllerGroup(x, y, z))`, call `SendableSpeedController.group(x, y, z)`.

File: src/main/java/frc/robot/maps/TempestMap.java (hunk 7 above)
Comment (msoucy) [reply src/main/java/frc/robot/maps/TempestMap.java]: It didn't autoformat because he disabled format on save...

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 27]
```diff
@@ -0,0 +1,296 @@
...
+    }
+    
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered(){
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(new SequentialCommandGroup(pierrePossesion(), runToClearBottomSensor()));
+         topIRStop.setDeadline(stopWhenBallsAtTop());
+         return topIRStop;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: If you want this to look pretty:

```java
return stopWhenBallsAtTop().deadlineWith(new SequentialCommandGroup(...));
```

I trust Ben's take on the logic though.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 28]
```diff
@@ -0,0 +1,296 @@
...
+    }
+    //This will stop all commands when the top IR sensor on Pierre is triggered
+
+    public CommandBase indexMotor(final double motorSpeed) {
+        return new StartEndCommand(() -> {
+            singulator.set(motorSpeed);
+        }, () -> {
+            singulator.set(0);
+        });
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Missing requirements

File: src/main/java/frc/robot/subsystems/Indexer.java (hunk 13 above)
Comment (msoucy) [reply src/main/java/frc/robot/subsystems/Indexer.java]: This is literally the same thing as `indexMotor(.75)`... shouldn't be necessary.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 29]
```diff
@@ -0,0 +1,296 @@
...
+    }
+
+    /*
+     * Command to run the singlator motor until the IR sensor 1 (analog 0) Command
+     * to run the singulator and pierre motor until the IR senor 1 is empty and IR
+     * senor 2 is covered Command to run the pierre motor until the IR sensor is
+     * uncovered
+     */
+    public CommandBase singulatorPossesion() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Use `FunctionalCommand`

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 30]
```diff
@@ -0,0 +1,296 @@
...
+                singulator.set(0);
+            }
+
+        };
+
+    }
+    //This command will make sure that the singulator has possesion of the ball
+
+    public CommandBase pierrePossesion() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Use `FunctionalCommand`.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 31]
```diff
@@ -0,0 +1,296 @@
...
+
+            }
+
+        };
+
+    }
+    // This command will make sure that pierre has possesion of the ball. It will be at the bottom
+
+    public CommandBase loadBallToTop() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Use `FunctionalCommand`

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 32]
```diff
@@ -0,0 +1,296 @@
...
+               
+            }
+
+        };
+
+    }
+    //this will bring the ball to the top
+    
+    public CommandBase unLoadBall() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: ,puɐɯɯoƆlɐuoᴉʇɔunℲ, ǝs∩

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 33]
```diff
@@ -0,0 +1,296 @@
...
+            }
+
+        };
+
+    }
+    //this will bring the ball to the shooter, it must already be at the top
+
+
+    public CommandBase runToClearBottomSensor() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: ,puɐɯɯoƆlɐuoᴉʇɔunℲ, ǝs∩

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 34]
```diff
@@ -15,33 +13,33 @@
...
+    public CommandBase runRoller() {
         return new StartEndCommand(() -> {
-            intakeMotor.set(INTAKE_MOTOR_SPEED);
-            deployPiston.set(Value.kForward);
+            intakeMotor.set(motorSpeed);
+
         }, () -> {
-            intakeMotor.stopMotor();
-            deployPiston.set(Value.kReverse);
+            intakeMotor.set(motorStop);
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Intake.java]: What do you have against `stopMotor()`?

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 35]
```diff
@@ -2,16 +2,42 @@
...
-    private SendableSpeedController shooterWheelMotor;
+    private SendableSpeedController shooterWheelMotor1;
+    private SendableSpeedController shooterWheelMotor2;
+
+    private double speed = -.85;
 
     public Shooter(RobotMap.ShooterMap map) {
         super();
-        shooterWheelMotor = map.shooterWheel();
+        shooterWheelMotor1 = map.shooterWheel1();
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Shooter.java]: This makes me think that they could probably just be a group?

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 36]
```diff
@@ -0,0 +1,237 @@
...
+
+            return !bottomPierreIR.getAsBoolean() || topPierreIR.getAsBoolean();
+
+        }, this);
+
+    }
+
+    // this will make space for another ball
+    public CommandBase stopWhenBallsAtTop() {
+        return new CommandBase() {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Use `WaitUntilCommand` here

File: src/main/java/frc/robot/Robot.java [hunk 37]
```diff
@@ -146,13 +170,14 @@ public SequentialCommandGroup passLine() {
...
      */
-
     private void configureButtonBindings() {
-        driveController.getButton(Button.kX).whenHeld(controlPanel.spinForwards());
-        driveController.getButton(Button.kY).whenHeld(controlPanel.spinBackwards());
-        copilotController.getButton(Button.kX).whenHeld(intake.intake());
-        copilotController.getButton(Button.kBumperRight).whenHeld(intake.discharge());
-        driveController.getButton(Button.kA).toggleWhenActive(
-                drive.drive(() -> -driveController.getTriggers(), () -> driveController.getX(Hand.kLeft)));
+        copilotController.getButton(Button.kA).whenHeld(singulatorAndIntake());
```
Comment (msoucy) [inline src/main/java/frc/robot/Robot.java:174]: Where did the "drive the robot" command go?

File: src/main/java/frc/robot/maps/RobotMap.java [hunk 38]
```diff
@@ -153,4 +93,34 @@ public LiftMap getLiftMap() {
         return new LiftMap();
     }
 
+    public static class IndexMap {
+        public SendableSpeedController singulator() {
```
Comment (msoucy) [inline src/main/java/frc/robot/maps/RobotMap.java:90]: Wait, the same functions are also in the `IntakeMap` above. What's up with that?

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 39]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    public SequentialCommandGroup shootOneBall() {
+        return new SequentialCommandGroup(loadBallToTop(), unLoadBall());
+    }
+
+    public ParallelDeadlineGroup topSensorTriggered() {
+        final ParallelDeadlineGroup topIRStop = new ParallelDeadlineGroup(
+                new SequentialCommandGroup(pierrePossesion(), runToClearBottomSensor()));
+        topIRStop.setDeadline(stopWhenBallsAtTop());
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Stylistic: Try using the `deadlineWith` method of a command.

```java
return stopWhenBallsAtTop().deadlineWith(new SequentialCommandGroup(pierrePossession(), runToClearBottomSensor()));
```

I personally think it looks nicer (but would look even nicer with `CommandGroupBase.sequence`)

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 40]
```diff
@@ -0,0 +1,220 @@
...
+
+        }, (interrupted) -> {
+
+            singulator.set(0);
+
+        }, () -> {
+
+            return frontIntakeIR.getAsBoolean() || topPierreIR.getAsBoolean();
+
+        }, this);
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: Try using `WaitUntilCommand`, which takes just the `isFinished` lambda, and then adding `andThen` afterwards to clear the singulator.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 41]
```diff
@@ -0,0 +1,220 @@
...
+
+    }
+    // This command will make sure that the singulator has possesion of the ball
+
+    public CommandBase pierrePossesion() {
+        return new FunctionalCommand(() -> {
+
+            if (frontIntakeIR.getAsBoolean()) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java]: This is also done in `execute`, might as well keep this empty. If there's really a good reason to do it, make it a temporary variable and pass it in twice.

File: src/main/java/frc/robot/subsystems/Indexer.java [hunk 42]
```diff
@@ -0,0 +1,220 @@
...
+            }
+            if (backIntakeIR.getAsBoolean()) {
+                pierreMotor.set(pierreIndexSpeed);
+                singulator.set(singulatorMotorSpeed);
+            }
+        }, (interrupted) -> {
+
+            pierreMotor.set(0);
+            singulator.set(0);
+            SmartDashboard.putNumber("Ball Count", ballCounting);
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Indexer.java:106]: Where is `ballCounting` set?

If this needs to be set all the time, override the `periodic` function and do it there.

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 43]
```diff
@@ -16,39 +13,33 @@
 // The intake needs to make sure the balls pass through it and don't get jammed(Member: Motor)
 
 public class Intake extends SubsystemBase {
-    private final SendableSpeedController intakeMotor;
-    private final IDSolenoid deployPiston;
+    private SendableSpeedController intakeMotor;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Intake.java]: This seems to undo the changes that are on master. Take the master version.

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Indexer.java (hunk 42 above)
Comment (bot190) [reply src/main/java/frc/robot/subsystems/Indexer.java:106]: We will deal with this in the future with OBLog

**Review by msoucy — APPROVED** [review]
LGTM :shipit:

**Review by bot190 — APPROVED** [review]
This looks good. We'll be back.
